test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingTestContext.cs

```diff
@@ -252,18 +252,18 @@ private void InitializeFiles()
             ProjectPath = Path.Combine(WorkingPath, "project.csproj");
 
             OutputPackagesPath = Path.Combine(WorkingPath, "packages");
-            CreateNuGetConfig(WorkingPath);
 
             Directory.CreateDirectory(OutputPackagesPath);
         }
 
-        private void CreateNuGetConfig(string workingDirectory)
+        internal void CreateNuGetConfig(string workingDirectory, string source)
         {
             string nugetConfigContent =
-                @"<?xml version='1.0' encoding='utf-8'?>
+                $@"<?xml version='1.0' encoding='utf-8'?>
 <configuration>
   <packageSources>
     <clear />
+    <add key=""http-feed"" value=""{source}"" allowInsecureConnections=""true""/>
```

Comment:
I really wish someone caught this in the PR that made this change in the first place.

This change is modifying the design of how these "test contexts" work. By default they're supposed to create an isolated environment for the test to run in. Since the test directory is in a `.test` subdirectory of the NuGet.Client repo, the test directory needs a nuget.config with `<clear/>` elements to avoid getting the NuGet.Client repo nuget.config package sources.

By removing the call to `CreateNuGetConfig` from the class constructor, it's now mandatory for every test using this class to now also call `CreateNuGetConfig` to meet the original design goals (and eliminate risks of downloading packages from public feeds that have a name clash with packages used by tests, and eliminate risks of writing a new push test that accidentally pushes to a real package source).  This violates the principal of least surprise, especially since `SimpleTestConext` has a similar name and before this PR had similar semantics.

Reply:
This class isn't using SimpleTestPathContext, which is probably one problem.

I think moving *all* of the config logic into the CachingTestContext is the right thing too: 

https://github.com/NuGet/NuGet.Client/pull/5858/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.csproj&show-viewed-files=true#diff-2110e5d1ccc026550e4eb2aa5c587597d79f23161b6a79c565e01c17879692f4R26-R31

already had a bunch of logic that probably should've been part of the constructor anyways. 

Any setters in that class should probably be removed and just passed in the constructor.